ansi/parser.go

```diff
@@ -151,57 +152,62 @@ func (p *Parser) advance(d ParserDispatcher, b byte, more bool) parser.Action {
 	if p.State != state {
 		switch p.State {
 		case parser.EscapeState:
-			p.performAction(d, parser.ClearAction, b)
+			p.performAction(d, parser.ClearAction, state, b)
+		}
+		switch state {
+		case parser.Utf8State:
+			// Clear the parser state if we're transitioning to the Utf8State.
+			p.performAction(d, parser.ClearAction, state, b)
 		}
```

Comment:
I don't see a need for a switch for something as simple

```suggestion
		if state == parser.Utf8State:
			// Clear the parser state if we're transitioning to the Utf8State.
			p.performAction(d, parser.ClearAction, state, b)
 		}
```